nav2_smac_planner/include/nav2_smac_planner/types.hpp

```diff
@@ -188,6 +188,18 @@ struct MotionPrimitive
   MotionPoses poses;
 };
 
+  /**
+   * @struct nav2_smac_planner::GoalState
+   * @brief A struct to store the goal state
+   */
+
+template<typename NodeT>
+struct GoalState
+{
+  NodeT * goal = nullptr;
```

Comment:
So we do `NodePtr &` in alot of places and while I don't see an issue with this implementation, it does trigger my defensive programming to not want to change things as they were occurring before. `_goal` was set as the output of `addToGraph` which returns a reference to the direct memory inserted in the data structure. `goal` now being coped into `GoalState` changes that behavior.

if we made this into a `NodeT * &` that would resolve it and the constructor would turn into `&goal` instead.